mod-audit-server/src/main/java/org/folio/util/marc/ParsedRecordUtil.java

```diff
@@ -0,0 +1,243 @@
+        userId = getUserIdFromMetadata(newRecord.getMetadata(), CREATED_BY);
+        recordType = newRecord.getRecordType();
+        difference = getDifference(newRecord.getParsedRecord(), event.getEventType());
+      }
+      case SOURCE_RECORD_UPDATED -> {
+        recordId = newRecord.getId();
+        userId = getUserIdFromMetadata(newRecord.getMetadata(), UPDATED_BY);
+        recordType = newRecord.getRecordType();
+        difference = calculateDifferences(oldRecord.getParsedRecord(), newRecord.getParsedRecord());
+      }
+      default -> {
+        recordId = oldRecord.getId();
+        userId = getUserIdFromMetadata(oldRecord.getMetadata(), UPDATED_BY);
+        recordType = oldRecord.getRecordType();
+        difference = getDifference(oldRecord.getParsedRecord(), event.getEventType());
+      }
+    }
+
+    return new MarcAuditEntity(
+      event.getEventId(),
+      event.getEventMetadata().getEventDate(),
+      recordId,
+      event.getEventMetadata().getPublishedBy(),
+      event.getEventType().getValue(),
+      userId,
+      recordType,
+      difference
+    );
+  }
+
+  private static String getUserIdFromMetadata(Map<String, Object> metadata, String key) {
+    return metadata.get(key).toString();
+  }
+
+  private static Map<String, Object> calculateDifferences(Map<String, Object> oldParsedRecord, Map<String, Object> newParsedRecord) {
+    var oldFields = flattenFields(oldParsedRecord);
+    var newFields = flattenFields(newParsedRecord);
+    return compareParsedRecords(oldFields, newFields);
+  }
+
+  private static Map<String, Object> getDifference(Map<String, Object> parsedRecord, SourceRecordDomainEventType type) {
+    List<Map<String, Object>> fields = new ArrayList<>();
+    flattenFields(parsedRecord).forEach((key, value) ->
+      fields.add(Map.of(FIELD_KEY, key, VALUE_KEY, value))
+    );
+    return type.equals(SourceRecordDomainEventType.SOURCE_RECORD_CREATED)
+      ? toMap(fields, null, null)
+      : toMap(null, fields, null);
+  }
+
+  /**
+   * Flattens the fields in the given parsed record map to a single-level map structure.
+   * The method processes the "content" key in the input map, extracts the field data,
+   * and formats it based on specified rules, combining fields with the same tag into lists when necessary.
+   *
+   * @param input a map representing a parsed record, expected to contain a "content" key
+   *              with relevant field data and metadata.
+   * @return a map where field tags are the keys and their corresponding values
+   * are formatted strings or lists of strings.
+   */
+  @SuppressWarnings("unchecked")
+  private static Map<String, Object> flattenFields(Map<String, Object> input) {
+    var content = getContent(input);
+    var fields = (List<Map<String, Object>>) content.get(FIELDS_KEY);
+    var result = new HashMap<String, Object>();
+    if (content.containsKey(LEADER_KEY)) {
+      result.put("LDR", content.get(LEADER_KEY));
```

Comment:
@dmytrokrutii  I think it would be better to extract "LDR" key to constant, to avoid coding issues in future.